.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+
+on:
```

Comment:
With the separation between creating a build and verifying stuff that we kind of established in #2, wouldn't it make sense to have different actions for pushes to `master` and PRs? PRs should be verified (linted, typechecked, etc.) but that doesn't necessarily need to happen on pushes to `master`.

At least as long as we don't just push things to `master` without a PR. Which sounds reasonable, though. We might even consider protecting the branch. :thinking:

Reply:
+1 to protecting the branch. But even so, I kinda like the master check because not all PR merges could be FF merges. And while the CI runs for PRs are run on the merge (I think?) I still like the idea of testing the actual master just to be sure :D

Reply:
Good point with the non-ff merges; although we could enforce that, of course. :thinking: (Actually **can** we enforce this on GH? :sweat_smile:) Then people would have to rebase or merge `master` back in before getting merged and the check would run that way.

I actually think that's not a bad idea, since while yes, when you check `master` pushes you see stuff breaking on `master`, but you still have broken stuff on `master` then. :thinking:

Reply:
No wait, that's BS, right? What does CI run on for a PR? Should it not run on `target_branch + pr_changes`?! :confused:

Reply:
Yeah as I said, I think the merge is tested in the PR CI run. So that should buy us some security already. And yeah, having broken stuff on `master` is meh, but it's better than having broken stuff on `master` and not noticing it! 

We can require PRs to be "up to date", but as we know, not everyone can rebase and merging master into that branch is meh.

Reply:
+1 for also checking `master`. We've seen in the past often enough that things that shouldn't happen still do happen.
And having these tests run on `master` doesn't really cost us anything as long as they are reliable.